frappe/public/js/frappe/ui/toolbar/search_utils.js

```diff
@@ -186,9 +186,64 @@ frappe.search.utils = {
 				route: route,
 			};
 		};
+
+		const pluralToSingular = {
```

Comment:
Try this instead? 

https://github.com/frappe/frappe/blob/05d7646ab2d72541418f0ddce04268187594d0b6/frappe/public/js/frappe/utils/utils.js#L38-L146

Reply:
C‘mon, this already existed? I‘m baffled it‘s not being used more. Thanks for bringing it up to me, @ankush!